dotnet/src/SemanticKernel/Planning/ConditionalFlowHelper.cs

```diff
@@ -0,0 +1,298 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// <para>Semantic skill that evaluates conditional structures</para>
+/// <para>
+/// Usage:
+/// var kernel = SemanticKernel.Build(ConsoleLogger.Log);
+/// kernel.ImportSkill("conditional", new ConditionalSkill(kernel));
+/// </para>
+/// </summary>
+public class ConditionalFlowHelper
+{
+    private readonly IKernel _kernel;
+
+    internal const string ReasonIdentifier = "Reason:";
+    internal const string NoReasonMessage = "No reason was provided";
+
+    private readonly ISKFunction _ifStructureCheckFunction;
+    private readonly ISKFunction _evaluateConditionFunction;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ConditionalFlowHelper"/> class.
+    /// </summary>
+    /// <param name="kernel"> The kernel to use </param>
+    /// <param name="completionBackend"> A optional completion backend to run the internal semantic functions </param>
+    internal ConditionalFlowHelper(IKernel kernel, ITextCompletion? completionBackend = null)
+    {
+        this._kernel = kernel;
+        this._ifStructureCheckFunction = kernel.CreateSemanticFunction(
+            ConditionalFlowConstants.IfStructureCheckPrompt,
+            skillName: "PlannerSkill_Excluded",
+            description: "Evaluate if an If structure is valid and returns TRUE or FALSE",
+            maxTokens: 100,
+            temperature: 0,
+            topP: 0.5);
+
+        this._evaluateConditionFunction = kernel.CreateSemanticFunction(
+            ConditionalFlowConstants.EvaluateConditionPrompt,
+            skillName: "PlannerSkill_Excluded",
+            description: "Evaluate a condition group and returns TRUE or FALSE",
+            maxTokens: 100,
+            temperature: 0,
+            topP: 0.5);
+
+        if (completionBackend is not null)
+        {
+            this._ifStructureCheckFunction.SetAIService(() => completionBackend);
+            this._evaluateConditionFunction.SetAIService(() => completionBackend);
+        }
+    }
+
+    /// <summary>
+    /// Get a planner if statement content and output <if/> or <else/> contents depending on the conditional evaluation.
+    /// </summary>
+    /// <param name="ifFullContent">Full statement content including if and else.</param>
+    /// <param name="context"> The context to use </param>
+    /// <returns>Then or Else contents depending on the conditional evaluation</returns>
+    /// <remarks>
+    /// This skill is initially intended to be used only by the Plan Runner.
+    /// </remarks>
+    public async Task<string> IfAsync(string ifFullContent, SKContext context)
```

Comment:
I haven't connected the dots in my head, but the method signature is basically an ISKFunction, so I'm wondering if we can't just use that syntax in the xml to enable the scenarios....